tests/conftest.py

```diff
@@ -34,15 +32,35 @@ def xvfb():
 
 @pytest.yield_fixture(scope='function')
 def xterm_window(xvfb):
-    with NamedTemporaryFile() as tf:
-        proc = subprocess.Popen(
-            ['xterm', '-T', 'My window title',
-             '-e', 'echo "$WINDOWID" > "{}"; bash'.format(tf.name)])
-        time.sleep(.100)
-        with open(tf.name, 'rb') as fp:
-            window_id = int(fp.read())
-    yield XtermProcessInfo(proc=proc, window_id=window_id)
-    proc.terminate()
+    """
+    Create an xterm window test fixture.  This fixture opens a new xterm
+    window and yields its the X window id.  Upon test completion the xterm
+    process is cleaned up.
+    :param xvfb:
+    :return:
+    """
+    directory = tempfile.mkdtemp()
```

Comment:
That's definitely a more elegant solution.  First time using py.test here. I was unaware this feature existed! The dependency injection that py.test uses for fixtures is certainly much different than unittest and nosetest.